bundles/automation/org.eclipse.smarthome.automation.core.test/src/test/java/org/eclipse/smarthome/automation/core/internal/RuleEngineTest.java

```diff
@@ -54,19 +50,21 @@ private RuleEngine createRuleEngine() {
     @Test
     public void testAddRetrieveRules() {
         RuleEngine ruleEngine = createRuleEngine();
-        Rule rule0 = new Rule();
+        Rule rule0 = ruleEngine.initRuleId(new Rule());
         ruleEngine.addRule(rule0, true);
-        Collection<Rule> rules = ruleEngine.getRules();
+        Collection<RuntimeRule> rules = ruleEngine.rules.values();
```

Comment:
Don't use rules.values. It should be private field. Do a method getRuntimeRules() which has to return unmodifiable collection of rules